96752745_Amal/assignment3.cu

```diff
@@ -0,0 +1,53 @@
+#include<iostream>
+using namespace std;
+
+__global__ void Transpose(int *d_a,int max){
+
+int i = blockIdx.x*blockDim.x+threadIdx.x;
+int j = blockIdx.y*blockDim.y+threadIdx.y;
+int id1 = i+max*j;
+int id2 = j+max*i;
+__syncthreads();
+
+if(i<max && j<max)

```

Comment:
You need to include another condition in the if statement to check if i<j (or i>j, any one of those). Because you need to operate only on one half of the matrix. You are swapping elements in one half with those in the other half, so only half the threads should perform this operation.